### DISCUSSION TITLE
Turn off map canvas rotation

### DISCUSSION THREAD

**Discussion by AK-survey** [opengisch/QField#4040] [Q&A]
Hello,

Seems like a really basic thing, but I would like to be able to turn off the map canvas rotation for v2.7.0. This is a handy feature when trying to navigate following a line or something like that, but it seems to rotate every last time I zoom in or out. I would much rather have a north-up orientation unless I specified otherwise. How do I change the settings to this? Nothing stood out to me in the settings or documentation.

**Comment by nirvn** [comment]
@AK-survey , what about increasing the threshold for rotation to occur instead of adding a cumbersome toggle? When you zoom in/out in say Google Maps, do you feel you also hit accidental rotation there or not?

**Reply by nirvn** [reply]
@AK-survey , follow up question: do you feel you're mostly triggering accidental rotation when zooming in or zooming out?

**Reply by nirvn** [reply]
@AK-survey , in 10 minutes or so some APKs will be up in this PR (https://github.com/opengisch/QField/pull/4045) for you to try. The rotation threshold has been increased to try and avoid accidental rotation.

**Reply by AK-survey** [reply]
A threshold value would be great, especially being able to customize it from settings. 10 degree increments would seem reasonable, with 0 degrees being no restriction and 90 degrees being no rotation ever. I found no difference with accidental rotation zooming in vs out. I just tried the PR 4045 beta and it worked great. As far as I'm concerned, I would stick with that threshold or make it the default if options are given.

**Reply by nirvn** [reply]
@AK-survey , good. Our initial threshold was tentative, and we weren't quite sure it was high enough. I think you've answered our question here :)

**Reply by aghiraldi** [reply]
I've often wondered how to block the orientation of the map. Because too often, it turns accidentally. Pressing north only reorients the map. The best thing would be to be able to deactivate this function.
